src/main/java/com/googlecode/download/maven/plugin/internal/HttpFileRequester.java

```diff
@@ -283,8 +304,11 @@ public void download(final File outputFile, List<Header> headers) throws IOExcep
      * @throws IOException thrown if I/O operations don't succeed
      */
     private Object handleResponse( URI uri, File outputFile, HttpCacheContext clientContext, HttpResponse response )
-            throws IOException
-    {
+            throws IOException, DownloadFailureException {
+        if (response.getStatusLine().getStatusCode() >= 300) {
```

Comment:
@ajarmoniuk shouldn't this be `>= 400` ? Why do we want to fail on 3xx ?

Reply:
Changed to >= 400.